packages/commercetools/composables/src/useProduct/index.ts

```diff
@@ -1,16 +1,38 @@
 import { UseProduct } from '@vue-storefront/interfaces'
 import { ref } from '@vue/composition-api'
+import { getProduct } from '@vue-storefront/commercetools-api'
+import { enhanceProduct } from './../helpers/internals'
+import { ProductVariant } from './../types/GraphQL'
 
-export default function useProduct (): UseProduct<any, any> {
+interface UseProductSearchParams {
+  slug?: string
+  catIds?: string[]
+  skus?: string[]
+}
+
+type Search = (params: UseProductSearchParams) => void
+
+const loadProductVariants = async (params: UseProductSearchParams): Promise<ProductVariant[]> => {
+  const productResponse = await getProduct(params)
+  const enhancedProductResponse = enhanceProduct(productResponse)
```

Comment:
Isn't better to define a type or interface? What do you think?

```TS
interface EnhanceProductResponse {
  data: {
    _variants: any
  }
}
```

Reply:
underscored fields are `private` for us. we decided to use them to avoid custom typing for eCommerce platforms. Instead of modifying response (that are already typed) we are adding new fields (underscored)